packages/cli/openapi-parser/src/loadOpenAPI.ts

```diff
@@ -79,11 +64,57 @@ export async function loadOpenAPI({
     }
 
     if (overridesFilepath != null) {
-        return await mergeWithOverrides<OpenAPI.Document>({
+        const merged = await mergeWithOverrides<OpenAPI.Document>({
             absoluteFilepathToOverrides: overridesFilepath,
             context,
             data: parsed
         });
+        // Run the merged document through the parser again to ensure that any override
+        // references are resolved.
+        return await parseOpenAPI({
+            absolutePathToOpenAPI,
+            parsed: merged
+        });
     }
     return parsed;
 }
+
+async function parseOpenAPI({
+    absolutePathToOpenAPI,
+    parsed
+}: {
+    absolutePathToOpenAPI: AbsoluteFilePath;
+    parsed?: OpenAPI.Document;
+}): Promise<OpenAPI.Document> {
+    const options: BundleOptions = {
+        config: new Config(
+            {
+                apis: {},
+                styleguide: {
+                    plugins: [FERN_TYPE_EXTENSIONS],
+                    rules: {
+                        spec: "warn"
+                    }
+                }
+            },
+            undefined
+        ),
+        dereference: false,
+        removeUnusedComponents: false,
+        keepUrlRefs: true
+    };
+    const result =
+        parsed != null
+            ? await bundle({
+                  ...options,
+                  doc: {
+                      source: new Source(absolutePathToOpenAPI, "<openapi>"),
```

Comment:
Will release this as an RC and come back to this - this fixes the issue at hand, but it's not immediately clear how `Source` is used and/or if it's necessary to serialize the parsed document as the body (instead of a `<openapi>` placeholder).